### DISCUSSION TITLE
Bug: "@typescript-eslint/no-extra-semi" and "semi" fight it out and add many extra semicolons (--fix)

### DISCUSSION THREAD

**Discussion by gwynjudd** [eslint/eslint#17190] [Q&A] [answered]
### Environment

PS C:\Users\gjudd\Documents\Develop\Mobility\Develop\Touch\Touch\core\app> npx eslint --env-info.
Environment Info:

Node version: v16.17.0
npm version: v8.15.0
Local ESLint version: Not found   
Global ESLint version: Not found  
Operating System: win32 10.0.19045
PS C:\Users\gjudd\Documents\Develop\Mobility\Develop\Touch\Touch\core\app> npx eslint --version  
v8.40.0
PS C:\Users\gjudd\Documents\Develop\Mobility\Develop\Touch\Touch\core\app> 

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

When I started writing this, I didn't actually realise I had both of these rules enabled. It's possible I'm doing something stupid, where I'm at now is the end of a sequence of logical actions.
```
/* eslint-env node */
module.exports = {
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:@typescript-eslint/recommended-requiring-type-checking',
	],
	plugins: ['@typescript-eslint'],
	parser: '@typescript-eslint/parser',
	parserOptions: {
		project: true,
		tsconfigRootDir: __dirname,
	},
	root: true,
	ignorePatterns: [
		".eslintrc.cjs",
		"vendor/*",
		"*.js",
		"*.d.ts"
	],
	rules: {
		"@typescript-eslint/no-extra-semi": "off",
		"semi": ["error", "always"],
	}
};

```
</details>

```ts
export default interface A {
	foo:number;
}
```

### What did you expect to happen?

I ran `eslint --fix .` in the working copy. I expected no changes in this file

### What actually happened?

The output file as follows

```ts
export default interface A {
	foo:number;
};;;;;;;;;;
```

### Link to Minimal Reproducible Example

https://github.com/gwynjudd/eslint-semi

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by kecrily** [comment, marked as answer]
In TypeScript, you shouldn't use the `semi` rules, but rather the `@typescript-eslint/semi` rules provided by typescript-eslint.

https://typescript-eslint.io/rules/semi/

**Reply by gwynjudd** [reply]
Thanks that makes sense
